src/index.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * Type for (x, y) coordinate pair.
+ */
+ export type Point = { x: number, y: number };
+
+ /**
+  * Method signature for distance calculation.
+  */
+ export type DistanceFunc = (p: Point, a: Point, b: Point) => number;
+ 
+ /**
+  * Distance and index as returned by `MaxDistance()`.
+  */
+ type DistanceIndex = { distance: number, index: number };
+ 
+ /**
+  * Calculates the perpendicular distance between point `p` and the line intersecting points `a` and `b`.
+  * @param p point
+  * @param a line point
+  * @param b line point
+  * @returns distance
+  */
+ export const PerpendicularDistance: DistanceFunc = (p: Point, a: Point, b: Point) => {
+     let distance;
+ 
+     // horizontal line
+     if (a.x === b.x) distance = Math.abs(p.x - a.x);
+ 
+     // vertical line
+     else if (a.y === b.y) distance = Math.abs(p.y - a.y);
+ 
+     // sloped line
+     else {
+         const slope = (b.y - a.y) / (b.x - a.x);
+         const intercept = a.y - (slope * a.x);
+         distance = Math.abs(slope * p.x - p.y + intercept) / Math.sqrt(Math.pow(slope, 2) + 1);
+     }
+ 
+     return distance;
+ }
+ 
+ /**
+  * Calculates the shortest distance between point `p` and the line segment between points `a` and `b`.
+  * @param p point
+  * @param a line point
+  * @param b line point
+  * @returns distance
+  */
+ export const ShortestDistance: DistanceFunc = (p: Point, a: Point, b: Point) => {
+ 
+     // function to calculate the square of the distance between two points
+     const pointDistSq = (i: Point, j: Point) => Math.pow(i.x - j.x, 2) + Math.pow(i.y - j.y, 2);
+ 
+     let distance;
+     const lineLenSq = pointDistSq(a, b);
```

Comment:
Codacy has a fix for the issue: [Delete `···`](https://app.codacy.com/gh/icooper/curvereduce-js/pullRequest?prid=9960097)
```suggestion
  const lineLenSq = pointDistSq(a, b);
```